### DISCUSSION TITLE
Rerun test suite on "Runner is not Alive"

### DISCUSSION THREAD

**Discussion by kskarra** [getgauge/gauge#1903] [General]
Heya!

Running a bunch of tests in parallel, and sometimes one of the runners crash, and gives a "Error Message: Runner is not Alive" on the end.
Is there some handy way to catch this (and/or specific errors), and order a rerun instead of the whole test run to fail? 
When manually rerunning the test afterwards, it usually passes fine.

**Comment by sriv** [comment]
Hi, There isn't a mechanism to retry on runner crash at the moment. Are you able to investigate a possible cause for the runner crash?

**Reply by kskarra** [reply]
Ok, thanks for coming back to me!
As for a cause I've no idea, and it's too sporadic to be debugged in a sensible way. Likely it's to webpage I'm testing towards that's behaving unexpectedly.

**Comment by KarianneM** [comment]
Hello,

I can reproduce this error, I get it every run. It happens sooner if the gauge.log file is quite full. The error I get along with the " runner is not alive" statement is "websocket is not open READY STATE 2/3" or : "Error: Connection to browser lost. This probably isn't a problem with Taiko, inspect logs for possible causes." After these errors the final message is "runner is not alive" . The stacktrace is in the Taiko discussions https://github.com/getgauge/taiko/discussions/1879

**Reply by sriv** [reply]
Thanks for this input. While the Taiko discussion has it's own analysis, I think gauge can handle this better. There was a suggestion to rotate the log files, to prevent it from becoming huge. Will take a shot at that.
